core/src/main/scala/org/bitcoins/core/protocol/ln/LnHumanReadablePart.scala

```diff
@@ -0,0 +1,67 @@
+package org.bitcoins.core.protocol.ln
+
+import org.bitcoins.core.config.{ MainNet, NetworkParameters, RegTest, TestNet3 }
+import org.bitcoins.core.protocol.ln.LightningNetworkPrefix._
+
+sealed abstract class LnHumanReadablePart {
+  def network: LightningNetworkParams
+
+  def amount: Option[LnCurrencyUnit]
+
+  override def toString: String = network.value + invoiceAmount
+
+  def invoiceAmount: String = {
+    if (amount.isDefined) {
+      amount.get.toBigInt.toString() + amount.get.character
+    } else ""
+  }
+
+  def bytes: Seq[Byte] = network.value.map(c => c.toByte) ++ invoiceAmount.map(c => c.toByte)
```

Comment:
We just modifed the api to have a `NetworkElement` return a `ByteVector` instead of a `Seq[Byte]`. This is because a `ByteVector` is MUCH more performant.

You should always use the `override` modifier to make sure you are following the interfaces specification. So this should look like 

```scala
override def bytes: ByteVector = { ... }
```